### PR TITLE
Added extra classes to include in the data wrapper#189

### PR DESCRIPTION
**Pull request by TheBonheurs**
- Relevant Issues: #170 
- Related Pull Requests: -

* * *

## What
This Pull Request adds to the repository 
Two extra classes that encapsylate collections of other classes.

## Why
This Pull Request is needed because 
We use a DataWrapper when we are sending data over bluetooth/nfc/qr and it encapsulates an object but won't work with a List. These new classes will save those lists within them.

## How
This feature can be viewed/tested within the project by 
Adding code to send multuple contacts or taps and seeing if they are parsed correctly.

## Alternative implementation
Other implementations that I've have considered are 
Using an arraylist (or any other list) as a param for DataWrapper but it proved to be cumbersome and produced nasty code...

## Notes

Not really

### CODE REVIEW

**Comment by TheBonheurs** [comment]
Sure! I was looking at the Contact tests as reference and it was missing there though 😨

**Comment by jverbraeken** [comment]
> Sure! I was looking at the Contact tests as reference and it was missing there though 😨 

Oops, fixed that in #185

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/tap/ATapData.java [hunk 1]
```diff
@@ -33,4 +33,29 @@ public Timestamp getTimestamp() {
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        
+        final ATapData ts = (ATapData) o;
+
+        return this.timestamp == null ? ts.timestamp == null : this.timestamp.equals(ts.timestamp);
```
Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/tap/ATapData.java]: Could you make this `equals` more like the one [in multicontacts](https://github.com/ericcornelissen/NervousFish/pull/190/files#diff-149fcabef3cefb35343ebbd60e7f63aeR38) 😇

Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/tap/ATapData.java]: Can you elaborate a bit more on this? What exactly should I change?

Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/tap/ATapData.java]: In what way do you want it to change? Which part exactly?

Comment (ericcornelissen) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/tap/ATapData.java]: Something like:
```java
public boolean equals(final Object o) {
    if (o == null || this.getClass() != o.getClass()) {
        return false;
    }
        
    final ATapData that = (ATapData) o;
    return this.timestamp == null ? that.timestamp == null : this.timestamp.equals(that.timestamp);
}
```

**Pull request merged**